### PR TITLE
NOJIRA 

### PR DESCRIPTION
**Pull request by axxter99**
Throwable -> Exception
Map => Map<<String, String> 
Collection<UserEdit>
..

### CODE REVIEW

File: providers/sample/src/java/org/sakaiproject/provider/authzGroup/SampleGroupProvider.java [hunk 1]
```diff
@@ -314,11 +314,11 @@ else if ("ta".equals(userId))
 	/**
 	 * {@inheritDoc}
 	 */
-	public Map getGroupRolesForUser(String userId)
+	public Map<String, String> getGroupRolesForUser(String userId)
 	{
 		update();
 
-		Map rv = new HashMap();
+		Map<String, String> rv = new HashMap<String, String>();
```
Comment (bjones86) [inline providers/sample/src/java/org/sakaiproject/provider/authzGroup/SampleGroupProvider.java]: You can use the diamond operator here

File: providers/sample/src/java/org/sakaiproject/provider/user/SampleUserDirectoryProvider.java [hunk 2]
```diff
@@ -299,9 +299,9 @@ public boolean findUserByEmail(UserEdit edit, String email)
 	 *        Use this factory's newUser() method to create all the UserEdit objects you populate and return in the return collection.
 	 * @return Collection (UserEdit) of user objects that have this email address, or an empty Collection if there are none.
 	 */
-	public Collection findUsersByEmail(String email, UserFactory factory)
+	public Collection<UserEdit> findUsersByEmail(String email, UserFactory factory)
 	{
-		Collection rv = new Vector();
+		Collection<UserEdit> rv = new Vector<UserEdit>();
```
Comment (bjones86) [inline providers/sample/src/java/org/sakaiproject/provider/user/SampleUserDirectoryProvider.java]: Diamond operator

Comment (axxter99) [inline providers/sample/src/java/org/sakaiproject/provider/user/SampleUserDirectoryProvider.java]: Collection<UserEdit> rv = new Vector<>;
?

Comment (bjones86) [inline providers/sample/src/java/org/sakaiproject/provider/user/SampleUserDirectoryProvider.java]:
```
Collection<UserEdit> rv = new Vector<>();
```

**Comment by bjones86** [comment]
Thanks, @axxter99; LGTM

**Comment by axxter99** [comment]
cheers!

**Pull request merged**